vividus-plugin-web-app-to-rest-api/src/main/java/org/vividus/crawler/LinkCrawler.java

```diff
@@ -71,12 +73,21 @@ public void visit(Page page)
 
     private static boolean isAllowedUrl(Page referringPage, WebURL url)
     {
-        return isNotFile(url) && isFromTheSameSite(referringPage.getWebURL().getURL(), url.getURL());
+        return isHtmlPage(url) && isFromTheSameSite(referringPage.getWebURL().getURL(), url.getURL());
     }
 
-    private static boolean isNotFile(WebURL url)
+    private static boolean isHtmlPage(WebURL url)
     {
-        return !StringUtils.endsWithAny(url.getPath().toLowerCase(), EXCLUDED_EXTENSIONS);
+        String fileExtensionFromUrl = FileNameUtils.getExtension(url.getPath().toLowerCase());
+        boolean htmlPage = fileExtensionFromUrl.isEmpty()
+                || ArrayUtils.contains(ALLOWED_EXTENSIONS, fileExtensionFromUrl);
+        if (!htmlPage)
+        {
+            LINK_CRAWLER_LOGGER.info(
+                    "URL \"{}\" refers to the file with unacceptable extension and will be excluded from crawling. "
```

Comment:
Won't this message look confusing?  Like something is wrong with the parsing. 
Something like: "Skipping crawling of URL {} "

Reply:
Imho it will be trashing the log with the messages :)

Reply:
> Imho it will be trashing the log with the messages :)

Agree. But it will help us to identify possible problems with the exclusion of correct HTML pages. I think it's worth simplify as in the your example above